examples/experimental/dagster-dlift/dagster_dlift/test/instance_fake.py

```diff
@@ -0,0 +1,46 @@
+from typing import Any, Mapping, NamedTuple
+
+from dagster_dlift.cloud_instance import DbtCloudInstance
+
+
+class ExpectedDiscoveryApiRequest(NamedTuple):
+    query: str
+    variables: Mapping[str, Any]
+
+    def __hash__(self) -> int:
+        return hash((self.query, frozenset(self.variables.items())))
+
+
+class ExpectedAccessApiRequest(NamedTuple):
+    subpath: str
+
+    def __hash__(self) -> int:
+        return hash(self.subpath)
+
+
+class DbtCloudInstanceFake(DbtCloudInstance):
+    """A version that allows users to fake API responses for testing purposes."""
+
+    def __init__(
+        self,
+        access_api_responses: Mapping[ExpectedAccessApiRequest, Any],
+        discovery_api_responses: Mapping[ExpectedDiscoveryApiRequest, Any],
```

Comment:
would we want these to be sequences (maybe `Mapping[ExpectedDiscoveryApiRequest, Sequence[Any]]`), so we could set up different responses for repeated queries? e.g. testing polling behavior or something

Reply:
yea I think I end up doing that in a downstream PR. But I'd want to couple it with the motivating use case